include/k4arecord/record.hpp

```diff
@@ -0,0 +1,280 @@
+/** \file record.hpp
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ * Kinect For Azure SDK - C++ wrapper.
+ */
+#ifndef K4A_RECORD_HPP
+#define K4A_RECORD_HPP
+
+#include <k4a/k4a.hpp>
+#include <k4arecord/record.h>
+
+namespace k4a
+{
+
+/** \class record record.hpp
+ * Wrapper for \ref k4a_record_t
+ *
+ * Wraps a handle for a record object
+ *
+ * \sa k4a_record_t
+ */
+class record
+{
+public:
+    /** Creates a k4a::record from a k4a_record_t
+     * Takes ownership of the handle, i.e. you should not call
+     * k4a_record_close on the handle after giving it to the
+     * k4a::record; the k4a::record will take care of that.
+     */
+    record(k4a_record_t handle = nullptr) noexcept : m_handle(handle) {}
+
+    /** Moves another k4a::record into a new k4a::record
+     */
+    record(record &&other) noexcept : m_handle(other.m_handle)
+    {
+        other.m_handle = nullptr;
+    }
+
+    record(const record &) = delete;
+
+    ~record()
+    {
+        flush();
+        close();
```

Comment:
Please add a comment that says flush is called on close so that we document what we learned.

Reply:
Is this comment correct?
```cpp
~record()
{
    // flush is called on closing record
    close();
}
```

Reply:
Yes :)